CavalcanteLucas/todolist
# Feature/OL41-todo_edit_title

**Link to task:** 
https://labcodes.atlassian.net/browse/OL-41

**How to test:**
With an user created and authenticated, create one or several TodoLists, and try to:
- Select TodoLists on the leftside of the screen (at the Action Bar), and check if detailed information about them appear on the main dashboard;
- After selecting a TodoList, click on the 'Edit button' besides it's title on the dashboard. It should open a dialog modal asking for a new title to that TodoList instance. Insert a new title and check if it's information has changed (both at the Action Bar as in the Dashboard).

**Description of your solution:**
This PR implements backend API and frontdend interfaces that enables TodoLists to be individually retrieved and updated. It comes with a major refactoring on the dashboard to match visual references on the Epic for this task. All the user's TodoLists are listed on the leftside of the page. Selecting each one of them will show it on the main dashboard. When updating a TodoList's title, alert messages appear on the topscreen of the dashboard to inform the user if it has been successful or if there was any errors. New tests introduced by this PR cover GET/PUT requests for the API endpoint to individual TodoLists, ensuring they are private etc. This PR also includes a small refactoring on the TodoList Serializer.
